app/src/main/java/ru/vssemikoz/newsfeed/MainActivity.java

```diff
@@ -77,10 +86,13 @@ public void onFailure(Call<NewsApiResponse> call, Throwable t) {
         };
     }
 
-    void initNewsItemListByResponse(Response<NewsApiResponse> response){
+    List<NewsItem> getNewsItemListByResponse(Response<NewsApiResponse> response){
+        ArrayList<NewsItem> newsItemList = new ArrayList<>();
```

Comment:
Try to avoid building your models or using concrete classes directly. You should use abstractions/interfaces. Consider this:
`List<NewsItem> news = new ArrayList<>();`
instead of this:
`ArrayList<NewsItem> newsItemList = new ArrayList<>();`